### DISCUSSION TITLE
Feature Request: Papagayo .dat file import for frame marker layers

### DISCUSSION THREAD

**Discussion by Cat-Butler** [keiwando/toonsquid#136] [Ideas]
Papagayo is a lipsync app (desktop only, unfortunately) that will attempt automatic breakdown of text (several languages) into phoneme (frame marker) keyframes, and help with easier manual alignment of frame marker phonemes (well, visemes really) corresponding to an audio (WAV) file.

Originally designed for Moho (desktop 2D animation software), there are importers and converters to create keyframes for several 2D and 3D animation systems.

The .dat format produced by Papagayo is a plain text file and will hopefully work well with ToonSquid frame markers:

MohoSwitch1
12 AI
13 L
15 O
16 WQ
18 MBP
20 rest

The first line can be ignored for import / alignment purposes (an identifier to check and differentiate any imported file from other non-Papagayo .dat formats), then it is just repeated lines of: 
[keyframe number] [space] [frame-marker label]

I'm hoping this is not too difficult to implement for importing as pre-aligned frame markers.

The original Papagayo app is here: https://lostmarble.com/papagayo/ but there are variations such as https://github.com/morevnaproject-org/papagayo-ng

Whether this gets implemented or not, I am grateful for such an elegant 2D animation app for ipad - thank you for ToonSquid, Keiwan!

**Comment by keiwando** [comment]
Thank you very much for the suggestion and the detailed description of the format - that is very helpful!
I will definitely look closer into this at some point in the future. Until then I'd love to hear how many other people also use Papagayo and would therefore benefit from this feature in ToonSquid.

**Comment by Cat-Butler** [comment]
Just a random thought that occurred to me recently:
I can easily write a converter (web-based, if required) for any text-based format.

So I'd also be happy with an input format of repeated lines of, for example: 
timeline frame, time (frame) number in symbol, as used in the Time slider property
e.g.
12  3
14  1
15  4
17  2
Or anything vaguely similar.

Note: other tools such as [Rhubarb](https://github.com/DanielSWolf/rhubarb-lip-sync) (even more automated) are also available, not just Papagayo. (Not for iPad, alas, only desktop.)
